src/main/java/org/sagebionetworks/bridge/spring/filters/RequestFilter.java

```diff
@@ -95,11 +111,23 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
         setRequestContext(builder.build());
 
         req = new RequestIdWrapper(request, requestId);
+
+        // set local cookies when a session in present
+        UserSession session = null;
         try {
             chain.doFilter(req, res);
+            session = (UserSession) request.getAttribute("CreatedUserSession");
+        } catch (ConsentRequiredException e) {
+            session = e.getUserSession();
```

Comment:
Similarly here. We don't need to catch the ConsentRequiredException, and the request.getAttribute() can be moved into the finally block.